tools/webpack/blocks.js

```diff
@@ -90,6 +92,10 @@ module.exports = [
 		plugins: [
 			...plugins,
 			new DependencyExtractionWebpackPlugin( { injectPolyfill: false } ),
+			new PhpFilePathsPlugin( {
+				context: './packages/block-library/src/',
```

Comment:
Should probably add the other paths used as `from` arg to the `CopyWebpackPlugin` instance as well.

Reply:
It's an internal plugin that will never be used outside so I don't think we need to spend more time on it if it works as is.

Reply:
Ah, I see what you did. It's also imported to use within the Gutenberg plugin. Hmm, that complicates it a little bit. In general, `@wordpress/scripts` should only offer scripts and default configs. We discourage using internal utils and extending configs as they are subject to change.

Reply:
What I meant with my comment is that the `CopyWebpackPlugin` initialization includes other directories that might contain blocks (not just `packages/block-library/src`, but also `packages/edit-widgets/src/blocks` and `packages/widgets/src/blocks`):

https://github.com/WordPress/gutenberg/blob/18164f0fcd09a282157f02bba7fd737012cc44fd/tools/webpack/blocks.js#L121-L128

AFAICS, it's just three blocks total across those two directories, but in theory, they _could_ at some point include a `render` or `variations` field in one of their `block.json`s, which is why I said we might want to include them in `PhpFilePathsPlugin` as well.

On the downside, this would make the logic in `PhpFilePathsPlugin` even more complicated 😕 So maybe it's okay to omit them.

---

> Ah, I see what you did. It's also imported to use within the Gutenberg plugin. Hmm, that complicates it a little bit. In general, `@wordpress/scripts` should only offer scripts and default configs. We discourage using internal utils and extending configs as they are subject to change.

Hmm, are you saying we shouldn't altogether be doing what we're doing in this PR? AFAICS, we have some instances where we're doing "deep" imports from `@wordpress/scripts` in other parts of GB, e.g. [this](https://github.com/WordPress/gutenberg/blob/18164f0fcd09a282157f02bba7fd737012cc44fd/test/e2e/playwright.config.ts#L11). (I thought deep imports might be fine, especially if the relevant symbols aren't exported publicly at package level?)

Reply:
Deep imports work perfectly fine. It’s manageable inside the Gutenberg monorepo as breaking changes have a high chance of bubbling up during the development. However it should be discouraged for usage through npm. In effect, we should never support requests to make it backward compatible on that level.

Reply:
> AFAICS, it's just three blocks total across those two directories, but in theory, they could at some point include a render or variations field in one of their block.jsons, which is why I said we might want to include them in PhpFilePathsPlugin as well.

Got it! Initially, I didn’t think about it because I assumed these changes apply exclusively to the `@wordpress/scripts` package. I personally would be in favor of landing changes in the current shape and try a follow up if you feel strongly about covering blocks located in different packages.

Reply:
No, I'd be more than happy to land this without further ado. If anything, I was trying to gauge if this has the potential of breaking things, but I guess that chance is pretty minimal.

I'll set the PR as ready for review.

Reply:
I'll try to give it a round of testing this week before approving it.